public/js/Cell.js

```diff
@@ -39,13 +42,19 @@ var Cell = function (Grid, ctx, row, col) {
                 ctx.fillStyle = "#0000FF";
                 break;
             case CellType.ARROW:
-                ctx.fillStyle = towerColors[0];
+                ctx.fillStyle = "#FFFFFF";
+                if(this.changed)
+                    this.img = towerImages[towerButtons.getLastButton()];
                 break;
             case CellType.LASER:
-                ctx.fillStyle = towerColors[1];
+                ctx.fillStyle = "#FFFFFF";
+                if(this.changed)
+                    this.img = towerImages[towerButtons.getLastButton()];
                 break;
             case CellType.ICE:
-                ctx.fillStyle = towerColors[2];
+                ctx.fillStyle = "#FFFFFF";
+                if(this.changed)
+                    this.img = towerImages[towerButtons.getLastButton()];
```

Comment:
Using this.changed will work for now while we're not doing much with drawing bullets or advanced creep movement (flying), but this will have to redraw the tower every frame eventually